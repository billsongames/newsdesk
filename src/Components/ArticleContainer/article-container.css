.article-container {
  display: flex;
  flex-direction: column;
}