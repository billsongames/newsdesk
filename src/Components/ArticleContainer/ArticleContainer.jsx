import React, { useEffect, useState } from "react";
import { supabase } from '../../api/api';
import axios from 'axios'

import Sidebar from "../Sidebar/Sidebar"
import ArticleCardMinor from "../ArticleCardMinor/ArticleCardMinor";
import Alert from "../Alert/Alert"

import "./article-container.css"

import { testData } from "../../data/data";
import APITest from "../APITest/APITest";

function ArticleContainer({userID, articleCategory, searchQuery}) {

  const testMode=true

  const [articles, setArticles] = useState([])
  const [savedArticles, setSavedArticles] = useState([])
  const [alert, setAlert] = useState({message: ""})
  const [reverse, setReverse] = useState(false);
  const [selected, setSelected] = useState("")
  const reverseOrder = () => {
    setReverse(!reverse);
  };
  const handleSelectChange = (selected) => {
    setSelected(selected);
    console.log(selected, "<selected");
  };
  let sortedArticles = articles;
  if (reverse) {
    sortedArticles = articles.slice().reverse();
  }
  let filteredArticles = sortedArticles;
  if (selected) {
    filteredArticles = sortedArticles.filter((article) => {
      return article.source.name === selected;
    });
  }



  
/*   useEffect(() => {
    async function getUserSavedArticles() {
      if (userID) {
        const {data, error} = await supabase
          .from('users')
          .select('saved_articles')
          .eq('user_id', `${userID}`)
    
          if (error) {
            console.log("error", error)
            setAlert({message: "Error retrieving saved articles"})
          } else {
            setSavedArticles(data[0].saved_articles)
            setAlert({message: ""})
          }
        }
    }
    getUserSavedArticles()

  }, [userID]) */

  useEffect(() => {
    if (testMode) {
      setArticles(testData.articles)
    }
  }, [testMode])


 // TOP HEADLINES ENDPOINT

/*   useEffect(() => {
    if (!searchQuery) {
      console.log(`No search query, ${articleCategory} category displayed`)
      axios
      .get(`https://gnews.io/api/v4/top-headlines?category=${articleCategory}&lang=en&country=gb&sortby=publishedAt&apikey=445b4b502608f3804329f4428b41b723`)

      .then(function (response) {
        setArticles(response.data.articles)
        setAlert({
          message: ""
        })

      })
      .catch(function (error) {
        setAlert({
          message: "ERROR RETRIEVING ARTICLES. PLEASE TRY LATER..."
        })
        console.log(error);
      })
    }

// SEARCH QUERY ENDPOINT
    
    else if (!articleCategory) {
      console.log(`Search query = ${searchQuery}, no category displayed`)

      axios
      .get(`https://gnews.io/api/v4/search?q=${searchQuery}&lang=en&country=us&max=10&apikey=445b4b502608f3804329f4428b41b723`)

      .then(function (response) {
        setAlert({
          message: ""
        })
        setArticles(response.data.articles)
        

      })
      .catch(function (error) {
        setAlert({
          message: "ERROR RETRIEVING ARTICLES. PLEASE TRY LATER..."
        })  
        console.log(error);
      })
    }
  
  } ,[articleCategory, searchQuery]) */









  return(
    <div className="article-container">
      <div>
        <Sidebar 
          reverseOrder={reverseOrder} 
          articles={articles}
          selected={selected}
          setSelected={handleSelectChange}
        />
      </div>
    
      <div>
        <Alert message={alert.message} />
        {filteredArticles.map((article) => (
          <div key={article.title}>
            <ArticleCardMinor
              title = {article.title}
              description={article.description}
              content={article.content}
              image={article.image}
              url={article.url}
              source={article.source.name}
              publishedAt={article.publishedAt}
              userID={userID}
/*               savedArticles={savedArticles} */
            />
        </div>
      ))}
      </div>  
  </div>
  )
}

export default ArticleContainer
