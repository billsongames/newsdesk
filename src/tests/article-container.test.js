import React from "react";
import { render } from "@testing-library/react";
import ArticleContainer from "../Components/ArticleContainer/ArticleContainer";

describe('Name of the group', () => {
  const validProps = {
    userID: jest.fn(), 
    articleCategory: jest.fn(),
    searchQuery: jest.fn()
  }

  it('renders correctly', () => {
    const { asFragment } = render(<ArticleContainer 
      userID={validProps.userID} 
      articleCategory={validProps.articleCategory} 
      searchQuery={validProps.searchQuery} 
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});


