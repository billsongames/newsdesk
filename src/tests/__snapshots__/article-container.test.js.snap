// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group renders correctly 1`] = `
<DocumentFragment>
  <div
    class="article-container"
  >
    <div>
      <div
        class="sidebar-container"
      >
        <div
          class="sidebar-filters-text"
        >
          Sort by Date:
          <button
            class="sidebar-button"
            data-testid="button1"
            disabled=""
          >
            Most Recent
          </button>
          <button
            class="sidebar-button"
            data-testid="button2"
          >
            Oldest
          </button>
        </div>
        <div
          class="sidebar-filters-text"
          data-testid="filter"
        >
          Filter by Source:
          <select>
            console.log()
            <option
              value=""
            >
              -All Sources-
            </option>
          </select>
        </div>
      </div>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
